src/Functions/parseTimeDelta.cpp

```diff
@@ -0,0 +1,243 @@
+
+        {"days", 24 * 3600},
+        {"day", 24 * 3600},
+        {"d", 24 * 3600},
+
+        {"hours", 3600},
+        {"hour", 3600},
+        {"hr", 3600},
+        {"h", 3600},
+
+        {"minutes", 60},
+        {"minute", 60},
+        {"min", 60},
+        {"m", 60},
+
+        {"seconds", 1},
+        {"second", 1},
+        {"sec", 1},
+        {"s", 1},
+    };
+
+    /** Prints amount of seconds in form of:
+     * "1 year 2 months 4 weeks 12 days 3 hours 1 minute 33 seconds".
+     * ' ', ';', '-', '+', ',', ':' can be used as separators, eg. "1yr-2mo", "2m:6s"
+     *
+     * The length of years and months (and even days in presence of time adjustments) are rough:
+     * year is just 365 days, month is 30.5 days, day is 86400 seconds, similarly to what formatReadableTimeDelta is doing.
+     */
+    class FunctionParseTimeDelta : public IFunction
+    {
+    public:
+        static constexpr auto name = "parseTimeDelta";
+        static FunctionPtr create(ContextPtr) { return std::make_shared<FunctionParseTimeDelta>(); }
+
+        String getName() const override { return name; }
+
+        bool isVariadic() const override { return true; }
+
+        bool isSuitableForShortCircuitArgumentsExecution(const DataTypesWithConstInfo & /*arguments*/) const override { return true; }
+
+        size_t getNumberOfArguments() const override { return 0; }
+
+        DataTypePtr getReturnTypeImpl(const DataTypes & arguments) const override
+        {
+            if (arguments.empty())
+                throw Exception(
+                    "Number of arguments for function " + getName() + " doesn't match: passed " + toString(arguments.size())
+                        + ", should be 1.",
+                    ErrorCodes::NUMBER_OF_ARGUMENTS_DOESNT_MATCH);
+
+            if (arguments.size() > 1)
+                throw Exception(
+                    "Number of arguments for function " + getName() + " doesn't match: passed " + toString(arguments.size())
+                        + ", should be at most 1.",
+                    ErrorCodes::NUMBER_OF_ARGUMENTS_DOESNT_MATCH);
+
+            const IDataType & type = *arguments[0];
+
+            if (!isString(type))
+                throw Exception("Cannot format " + type.getName() + " as time string", ErrorCodes::ILLEGAL_TYPE_OF_ARGUMENT);
+
+            return std::make_shared<DataTypeFloat64>();
+        }
+
+        bool useDefaultImplementationForConstants() const override { return true; }
+
+        ColumnPtr executeImpl(const ColumnsWithTypeAndName & arguments, const DataTypePtr &, size_t input_rows_count) const override
+        {
+            auto col_to = ColumnFloat64::create();
+            auto & res_data = col_to->getData();
+
+            for (size_t i = 0; i < input_rows_count; ++i)
+            {
+                std::string_view str{arguments[0].column->getDataAt(i)};
+                Int64 length = str.length();
+                Int64 fast = 0;
+                Int64 slow = 0;
```

Comment:
Not so good names for indexes, it's hard to understand their meaning, maybe you can use better naming